packages/odyssey-react-mui/src/ui-shell/renderUiShell.tsx

```diff
@@ -0,0 +1,105 @@
+/*!
+ * Copyright (c) 2024-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { type SetStateAction } from "react";
+import { ErrorBoundary } from "react-error-boundary";
+
+import { bufferUntil } from "./bufferUntil";
+import { createMessageBus } from "./createMessageBus";
+import { UiShell, UiShellProps, type UiShellComponentProps } from "./UiShell";
+import { renderReactInWebComponent } from "../web-component/renderReactInWebComponent";
+
+export const optionalComponentSlotNames: Record<
+  keyof Required<UiShellProps>["optionalComponents"],
+  string
+> = {
+  additionalTopNavItems: "additional-top-nav-items",
+  footer: "footer",
+  logo: "logo",
+  searchField: "search-field",
+};
+
+export const renderUiShell = ({
+  appRootElement: explicitAppRootElement,
+  onError = console.error,
+  uiShellRootElement,
+}: {
+  appRootElement?: HTMLDivElement;
```

Comment:
Similar to an earlier comment, since this is effectively the public API we're wanting folks to use, it'd be nice if these types had some doc to help define what they are. The number of consumers of this are small, but it'd still be good to be verbose / descriptive.

Reply:
f2a373f3391a6bfbfece4d6f6eced735b7a2508e